recipes/repo.rb

```diff
@@ -127,11 +127,14 @@
   when 'debian'
     apt_domain_name = 'public-repo-1.hortonworks.com'
     apt_base_url = "http://#{apt_domain_name}/HDP"
-    # HDP only supports Debian 6 and Ubuntu 12
-    case node['platform']
-    when 'debian'
+    # HDP supports Debian 6, Ubuntu 12. Starting with 2.3.2.0 Ubuntu 14 and Debian 7
+    if node['platform'] == 'debian' && node['platform_version'] =~ /^7/ && Gem::Version.new(hdp_update_version) >= Gem::Version.new('2.3.2.0')
```

Comment:
It's possible for `hdp_update_version` to be nil here. You need to move the `hdp_update_version` assignment at ~137 to above this case, since you're now using that variable in this block, rather than after it.